fe1-web/src/features/rollCall/components/RollCallOpen.tsx

```diff
@@ -0,0 +1,175 @@
+import { ToolbarItem } from 'core/components/Toolbar';
+import { AppParamList } from 'core/navigation/typing/AppParamList';
+import { LaoEventsParamList } from 'core/navigation/typing/LaoEventsParamList';
+import { LaoParamList } from 'core/navigation/typing/LaoParamList';
+import { Hash, PublicKey, Timestamp } from 'core/objects';
+import { ScannablePopToken } from 'core/objects/ScannablePopToken';
+import { Typography } from 'core/styles';
+import { FOUR_SECONDS } from 'resources/const';
+import STRINGS from 'resources/strings';
+
+import { RollCallHooks } from '../hooks';
+import { requestCloseRollCall } from '../network';
+import { RollCall } from '../objects';
+import AttendeeList from './AttendeeList';
+
+type NavigationProps = CompositeScreenProps<
+  StackScreenProps<LaoEventsParamList, typeof STRINGS.navigation_lao_events_view_single_roll_call>,
+  CompositeScreenProps<
+    StackScreenProps<LaoParamList, typeof STRINGS.navigation_lao_events>,
+    StackScreenProps<AppParamList, typeof STRINGS.navigation_app_lao>
+  >
+>;
+
+const RollCallOpen = ({ rollCall, laoId, isOrganizer, scannedPopTokens }: IPropTypes) => {
+  const generateToken = RollCallHooks.useGenerateToken();
+  const hasSeed = RollCallHooks.useHasSeed();
+  const toast = useToast();
+  const navigation = useNavigation<NavigationProps['navigation']>();
+
+  const [popToken, setPopToken] = useState('');
+  const [hasWalletBeenInitialized, setHasWalletBeenInitialized] = useState(hasSeed());
+
+  const onAddAttendees = useCallback(() => {
+    // Once the roll call is opened the first time, idAlias is defined
+    if (rollCall.idAlias) {
+      navigation.navigate(STRINGS.navigation_app_lao, {
+        screen: STRINGS.navigation_lao_events,
+        params: {
+          screen: STRINGS.navigation_lao_events_open_roll_call,
+          params: {
+            rollCallId: rollCall.id.toString(),
+            attendeePopTokens: (scannedPopTokens || []).map((e) => e.valueOf()),
+          },
+        },
+      });
+    } else {
+      toast.show(STRINGS.roll_call_location_error_scanning_no_alias, {
+        type: 'danger',
+        placement: 'bottom',
+        duration: FOUR_SECONDS,
+      });
+      console.debug(STRINGS.roll_call_location_error_scanning_no_alias);
+    }
+  }, [toast, navigation, rollCall, scannedPopTokens]);
+
+  const onCloseRollCall = useCallback(async () => {
+    // get the public key as strings from the existing rollcall
+    const previousAttendees = (rollCall.attendees || []).map((key) => key.valueOf());
+    // add the create a set of all attendees (takes care of deduplication)
+    const allAttendees = new Set([...previousAttendees, ...(scannedPopTokens || [])]);
+    // create PublicKey instances from the set of strings
+    const attendeesList = [...allAttendees].map((key: string) => new PublicKey(key));
+
+    if (!rollCall.idAlias) {
+      throw new Error('Trying to close a roll call that has no idAlias defined');
```

Comment:
This is pretty much the same guard clause. 
I'm wondering if we should put a pretty message here as well 😁